### PR TITLE
fix card state

### PR DESCRIPTION
**Pull request by shirmagen**

### CODE REVIEW

File: examples/client/Locomotion/src/context/payments/index.js [hunk 1]
```diff
@@ -43,17 +43,18 @@ const usePayments = () => {
...
   const clientHasValidPaymentMethods = () => paymentMethods.length > 0 && paymentMethods.some(pm => !pm.isExpired);
+  const isCashPaymentEnabled = () => getByKey(SETTINGS_KEYS.CASH_ENABLED);
 
   const getClientDefaultMethod = () => {
     if (paymentMethods && paymentMethods.length) {
       return (paymentMethods || []).find(pm => pm.isDefault) || paymentMethods[0];
-    } if (isCashPaymentEnabled) {
+    }
+    const cashEnabled = isCashPaymentEnabled();
+    if (cashEnabled === 'true') {
```
Comment (ofek-autofleet) [inline examples/client/Locomotion/src/context/payments/index.js]: isnt it boolean?

File: examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/index.tsx [hunk 2]
```diff
@@ -44,6 +42,32 @@ const RideOptions = () => {
     setPopupToShow(null);
   };
 
+
+  useEffect(() => {
+    const updateClient = async () => {
```
Comment (ofek-autofleet) [inline examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/index.tsx:48]: why do you define the function inside of the use effect

Comment (ofek-autofleet) [reply examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/index.tsx:48]: ?

File: examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/index.tsx [hunk 3]
```diff
@@ -44,6 +42,32 @@ const RideOptions = () => {
     setPopupToShow(null);
   };
 
+
+  useEffect(() => {
```
Comment (ofek-autofleet) [inline examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/index.tsx:47]: the use effect doesnt have a second parameter

**Review by ofek-autofleet — CHANGES_REQUESTED** [review]

File: examples/client/Locomotion/src/context/payments/index.js [hunk 4]
```diff
@@ -43,17 +43,18 @@ const usePayments = () => {
...
 
   const clientHasValidPaymentMethods = () => paymentMethods.length > 0 && paymentMethods.some(pm => !pm.isExpired);
+  const isCashPaymentEnabled = () => getByKey(SETTINGS_KEYS.CASH_ENABLED);
 
   const getClientDefaultMethod = () => {
     if (paymentMethods && paymentMethods.length) {
       return (paymentMethods || []).find(pm => pm.isDefault) || paymentMethods[0];
-    } if (isCashPaymentEnabled) {
+    }
+    const cashEnabled = isCashPaymentEnabled();
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/context/payments/index.js]: await

File: examples/client/Locomotion/src/pages/Payments/index.js [hunk 5]
```diff
@@ -21,27 +21,19 @@ export default ({ navigation, menuSide }) => {
   } = usePayments;
   const [pageLoading, setPageLoading] = useState(true);
   const [loading, setLoading] = useState(false);
-  let hasPaymentMethods = paymentMethods && paymentMethods.length > 0;
-  const [showList, setShowList] = useState(hasPaymentMethods);
+  const hasPaymentMethods = paymentMethods && paymentMethods.length > 0;
+  const [showList, setShowList] = useState(true);
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/Payments/index.js]: change to hasPaymentMethods

File: examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/index.tsx [hunk 6]
```diff
@@ -89,6 +113,9 @@ const RideOptions = () => {
         }}
       />
       <ChoosePaymentMethod
+        selected={ride?.paymentMethodId?.length
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/ActiveRide/RideDrawer/RideOptions/index.tsx:116]: why do u need .length?
I think usePayments.paymentMethods.includes(ride?.paymentMethodId) is enough

File: examples/client/Locomotion/src/pages/AuthScreens/AuthLoadingScreen.js [hunk 7]
```diff
@@ -46,7 +46,7 @@ const AuthLoadingScreen = ({ navigation }) => {
     async function getFromStorage() {
       const clientProfile = await StorageService.get('clientProfile');
       if (clientProfile) {
-        await versionCheck();
+        // await versionCheck();
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/AuthScreens/AuthLoadingScreen.js]: ?

**Review by avivbarniv — APPROVED** [review]

**Pull request merged**